portfolio-project-matching-app/backend/daoTechnology.js

```diff
@@ -0,0 +1,54 @@
+import {
+    getCollectionSnapshot,
+    getDocSnapshotById,
+} from '../Firebase/clientApp.ts'
+import { Technology } from '../models/Technology'
+
+/*
+    READ
+*/
+
+const getAllTechnologies = async () => {
```

Comment:
These two `READ` type functions are ones that we will most likely want to use, I haven't quite thought of a clever way to modularize these functions without simply passing the document snapshot straight to the calling function (which could potentially work, but would lose the association arrays).